### PR TITLE
feat(test): adding test

### PR DESCRIPTION
**Pull request by Sma1lboy**
#3110

### CODE REVIEW

File: ee/tabby-webserver/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 mod axum;
 mod hub;
 mod jwt;
+mod mock;
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/lib.rs:5]:
```suggestion
mod testutils;
```

File: ee/tabby-webserver/src/mock/mod.rs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    doc::{DocSearch, DocSearchDocument, DocSearchError, DocSearchHit, DocSearchResponse},
+};
+use tabby_inference::ChatCompletionStream;
+use tabby_schema::{
+    context::{ContextInfo, ContextService},
+    policy::AccessPolicy,
+    Result,
+};
+
+pub struct MockChatCompletionStream;
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/mock/mod.rs:23]:
```suggestion
pub struct FakeChatCompletionStream;
```

Usually we call an implementation `Mock` if it's behavior can be defined in testing time (e.g mockall: https://docs.rs/mockall/latest/mockall/).

For implementation with fixed output (for testing), usually we call it `FakeXXXX`

File: ee/tabby-webserver/src/service/answer.rs [hunk 3]
```diff
@@ -484,28 +484,180 @@ Remember, don't blindly repeat the contexts verbatim. When possible, give code s
...
+        },
+    };
+
+    lazy_static::lazy_static! {
+        static ref MOCK_CHAT: Arc<dyn ChatCompletionStream> = Arc::new(MockChatCompletionStream);
+        static ref MOCK_CODE: Arc<dyn CodeSearch> = Arc::new(MockCodeSearch);
+        static ref MOCK_DOC: Arc<dyn DocSearch> = Arc::new(MockDocSearch);
+        static ref MOCK_CONTEXT: Arc<dyn ContextService> = Arc::new(MockContextService);
+        static ref MOCK : Option<Box<dyn DocSearch>> = Some(Box::new(MockDocSearch));
+    }
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/answer.rs:515]:
```suggestion
```

I recommend just call `Arc::new` in place used these Fake implementations.

File: ee/tabby-webserver/src/service/answer.rs [hunk 4]
```diff
@@ -484,28 +484,180 @@ Remember, don't blindly repeat the contexts verbatim. When possible, give code s
...
+
+    #[test]
+    fn test_new() {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/answer.rs:518]: no need to test config equal ness

**Comment by Sma1lboy** [comment]
move to #3180 with tabby branch

**Pull request closed without merging**